spec/models/user_spec.rb

```diff
@@ -47,26 +47,77 @@
   describe '.from_token' do
     subject(:user) { described_class.from_token(token: UserProfileMock::TOKEN) }
 
-    let(:owner) { create(:owner, school:, name: 'School Owner', email: 'school-owner@example.com') }
+    context 'when logged into a full account' do
+      let(:owner) { create(:owner, school:, name: 'School Owner', email: 'school-owner@example.com') }
 
-    before do
-      authenticated_in_hydra_as(owner)
-    end
+      before do
+        authenticated_in_hydra_as(owner)
+      end
 
-    it 'returns an instance of the described class' do
-      expect(user).to be_a described_class
-    end
+      it 'returns an instance of the described class' do
+        expect(user).to be_a described_class
+      end
 
-    it 'returns a user with the correct ID' do
-      expect(user.id).to eq owner.id
-    end
+      it 'returns a user with the correct ID' do
+        expect(user.id).to eq owner.id
+      end
 
-    it 'returns a user with the correct name' do
-      expect(user.name).to eq 'School Owner'
+      it 'returns a user with the correct name' do
+        expect(user.name).to eq owner.name
+      end
+
+      it 'returns a user with the correct email' do
+        expect(user.email).to eq 'school-owner@example.com'
+      end
+
+      it 'returns a user without a username' do
+        expect(user.username).to be_nil
+      end
+
+      context 'when BYPASS_OAUTH is true' do
+        around do |example|
+          ClimateControl.modify(BYPASS_OAUTH: 'true') do
+            example.run
+          end
+        end
+
+        it 'does not call the API' do
+          user
+          expect(WebMock).not_to have_requested(:get, /.*/)
+        end
+
+        it 'returns a stubbed user' do
+          expect(user.name).to eq('School Owner')
+        end
+      end
     end
 
-    it 'returns a user with the correct email' do
-      expect(user.email).to eq 'school-owner@example.com'
+    context 'when logged into a student account' do
+      let(:student) { create(:student, school:, name: 'School Student') }
+
+      before do
+        authenticated_in_hydra_as(student)
+      end
+
+      it 'returns an instance of the described class' do
+        expect(user).to be_a described_class
+      end
+
+      it 'returns a user with the correct ID' do
+        expect(user.id).to eq student.id
```

Comment:
it's not obvious to me that the id that's being passed into to the model is starting with a `student:` prefix, assuming we do need the logic in the model to handle this (which I'm sure we do) it would be good to add an explicit test around this, unless I'm missing something implicit!

Reply:
no you're right, I don't actually think the id here is prefixed!